src/models/preference/write.js

```diff
@@ -0,0 +1,34 @@
+import sql from 'sql-template-tag'
+import { app } from '../../index.js'
+
+export const upsert = async (identityId, preferences, { transaction = null } = {}) => {
+  const client = transaction ?? app.db
+
+  const commitedPreferences = []
+
+  for (const preference of preferences) {
```

Comment:
@mohammadhb no need to use this query on the loop with `EXCLUDED` variale you can handle conflict values

Reply:
@jeyem resolved and tested